### ISSUE TITLE
Adding numStartreeUsed in query stats

### ISSUE THREAD

**Issue by fx19880617** [apache/incubator-pinot#6487]
This is to provide more insights on how many segments are using star-tree for every query.

**Comment by fx19880617** [comment]
cc: @Jackie-Jiang

**Comment by Jackie-Jiang** [comment]
Is this mostly for default star-tree where we don't know if a dimension is included?

**Comment by fx19880617** [comment]
Yes, this is a good sign for users to check if the query is fitting for star-tree or not.
This will also help when we are adding a new star-tree, we can just query to see the rebuild progress

**Comment by harishks** [comment]
Will be happy to pick this one up.

**Comment by vijaya-lakshmi-venkatraman** [comment]
Hello,
I would like to give a try if this is still available.
